### ISSUE TITLE
[exporter/azuredataexplorerexporter] Build error due to `github.com/Azure/azure-kusto-go`

### ISSUE THREAD

**Issue by gr0l0rg** [open-telemetry/opentelemetry-collector-contrib#18517]
## Description

It seems that the package `github.com/Azure/azure-kusto-go v0.9.2` which is imported by the exporter `github.com/open-telemetry/opentelemetry-collector-contrib/exporter/azuredataexplorerexporter` causes error while building or testing

## Steps to Reproduce

write your own collector that uses the exporter in `github.com/open-telemetry/opentelemetry-collector-contrib/exporter/azuredataexplorerexporter`

## Expected Result

successful building/testing

## Actual Result

Building causes the following errors

```
# github.com/Azure/azure-kusto-go/kusto/ingest/internal/queued
~/go/pkg/mod/github.com/!azure/azure-kusto-go@v0.9.2/kusto/ingest/internal/queued/queued.go:49:59: undefined: azblob.BlockBlobClient
~/go/pkg/mod/github.com/!azure/azure-kusto-go@v0.9.2/kusto/ingest/internal/queued/queued.go:49:83: undefined: azblob.UploadStreamToBlockBlobOptions
~/go/pkg/mod/github.com/!azure/azure-kusto-go@v0.9.2/kusto/ingest/internal/queued/queued.go:49:123: undefined: azblob.BlockBlobCommitBlockListResponse
~/go/pkg/mod/github.com/!azure/azure-kusto-go@v0.9.2/kusto/ingest/internal/queued/queued.go:52:56: undefined: azblob.BlockBlobClient
~/go/pkg/mod/github.com/!azure/azure-kusto-go@v0.9.2/kusto/ingest/internal/queued/queued.go:52:80: undefined: azblob.HighLevelUploadToBlockBlobOption
~/go/pkg/mod/github.com/!azure/azure-kusto-go@v0.9.2/kusto/ingest/internal/queued/queued.go:63:25: undefined: azblob.TransferManager
~/go/pkg/mod/github.com/!azure/azure-kusto-go@v0.9.2/kusto/ingest/internal/queued/queued.go:86:75: undefined: azblob.BlockBlobClient
~/go/pkg/mod/github.com/!azure/azure-kusto-go@v0.9.2/kusto/ingest/internal/queued/queued.go:86:107: undefined: azblob.UploadStreamToBlockBlobOptions
~/go/pkg/mod/github.com/!azure/azure-kusto-go@v0.9.2/kusto/ingest/internal/queued/queued.go:273:49: undefined: azblob.ContainerClient
~/go/pkg/mod/github.com/!azure/azure-kusto-go@v0.9.2/kusto/ingest/internal/queued/queued.go:325:84: undefined: azblob.ContainerClient
~/go/pkg/mod/github.com/!azure/azure-kusto-go@v0.9.2/kusto/ingest/internal/queued/queued.go:86:107: too many errors
```

### Collector version

v0.71.0

## Environment
OS: (e.g., "Ubuntu 20.04")
Compiler(if manually compiled): (e.g., "go 14.2")

### Additional context

This error happened in older versions also (v0.63.0)

**Comment by asaharn** [comment]
Hi @gr0l0rg , 

We have checked out the code and rebuilt without encountering any errors.

![image](https://user-images.githubusercontent.com/102726227/218394988-563a7c91-e576-441e-8286-f7f9139a9195.png)

We would appreciate your assistance in addressing the following queries to gain a deeper understanding of the issue:
- Is there a particular reason for building from source?
- What version of Go was used for the build? It has to be "go 1.18"
- On which operating system is the build taking place?

We would be happy to provide further assistance if you have any additional questions.

**Comment by asaharn** [comment]
Should have fixed with #19136
